test/cache/BrowserBasedPackageCache.test.ts

```diff
@@ -0,0 +1,202 @@
+        'fhir.small.wrong#0.1.0'
+      );
+      // Check that package JSON is added
+      const packageJSON = allSavedResources.find(
+        (r: any) => r.resourceType === 'packagejson' && r.id === 'fhir.small.wrong'
+      );
+      expect(packageJSON.name).toEqual('fhir.small.wrong');
+      expect(packageJSON.version).toEqual('0.1.0');
+      // Check that resources in example and other folders are added if possible
+      const patientExample = allSavedResources.find(
+        (r: any) => r.resourceType === 'Patient' && r.id === 'PatientExample'
+      );
+      expect(patientExample.name).toBeDefined();
+      const bundle = allSavedResources.find((r: any) => r.resourceType === 'Bundle');
+      expect(bundle).toBeUndefined(); // Bundle resource doesn't have an id
+    });
+
+    // Current implementation of BrowserBasedPackageCache does not overwrite package if it exists
+    // This is a difference from the DiskBasedPackageCache but has not been needed yet
+    it.skip('should cache a fhir package tarball to the cache folder and replace the existing cached package if applicable', async () => {});
+  });
+
+  describe('#isPackageInCache', () => {
+    it('should return true for a package in the cache', () => {
+      expect(cache.isPackageInCache('fhir.small', '0.1.0')).toBeTruthy();
+    });
+
+    it('should return false for a package not in the cache', () => {
+      expect(cache.isPackageInCache('bug', '0.1.0')).toBeFalsy();
+    });
+
+    it('should return false for a package with different version in the cache', () => {
+      expect(cache.isPackageInCache('fhir.small', '0.2.0')).toBeFalsy();
+    });
+  });
+
+  describe('#getPackagePath', () => {
+    it('should return the package path for a package in the cache', () => {
+      expect(cache.getPackagePath('fhir.small', '0.1.0')).toBe('fhir.small#0.1.0');
+    });
+
+    it('should return undefined for a package not in the cache', () => {
+      expect(cache.getPackagePath('big', '0.1.0')).toBeUndefined();
+    });
+
+    it('should return undefined for a package with different version in the cache', () => {
+      expect(cache.getPackagePath('fhir.small', '0.2.0')).toBeUndefined();
+    });
+  });
+
+  describe('#getPackageJSONPath', () => {
+    it('should return the path to package.json for a package in the cache', () => {
+      expect(cache.getPackageJSONPath('fhir.small', '0.1.0')).toBe(
+        'fhir.small#0.1.0#packagejson-fhir.small'
+      );
+    });
+
+    it('should return undefined for a package not in the cache', () => {
+      expect(cache.getPackageJSONPath('big', '0.1.0')).toBeUndefined();
+    });
+
+    it('should return undefined for a package with different version in the cache', () => {
+      expect(cache.getPackageJSONPath('fhir.small', '0.2.0')).toBeUndefined();
+    });
+  });
+
+  describe('#getPotentialResourcePaths', () => {
+    // Note: This implementation differs from DiskBasedPackageCache because it returns potentials from nested folders
```

Comment:
Why? Is this for legacy reasons (i.e., that's how FSH Online has always worked)?

Reply:
I do think it aligns with the legacy approach, but this was actually something I was a little surprised about when I wrote the test, but didn't change it because of the legacy reasons. But the reason it differs from the DiskBasedPackageCache is because the `getPotentialResourcePaths` function there does a `fs.readdirSync` that does not use the `recursive: true` option, so even though the files are in the package folder in the cache, they aren't returned as a potential path. In the BrowserBasedPackageCache, we just store all the resources without any notion of the folder they were from, so when we get the potential paths, we get the potential paths of resources that would have come from a nested folder. Rather than trying to change how we extract and store the resources, I decided to just note the difference in the test since that's how FSH Online has been operating. However, I'm happy to look into not storing the resources when we cache the package.

Reply:
Ideally, the behavior across implementations (DiskBased and BrowserBased) would be the same. If it's a fairly simple change, I think it's worth doing (and might save us some space in indexeddb). But... given the time (Dec 31), I think it's OK to leave as-is if the fix is not simple.

Reply:
[f64fc81](https://github.com/FHIR/fhir-package-loader/pull/52/commits/f64fc81c2b5564d80cf91d6b7a844d191c8a7fe9) aligns the two. The only way I saw to check if a file was in a nested folder was to check if there are multiple portions to the `header.name` property (so `package/StructureDefinition-example.json` is allowed but `package/other/StructureDefinition-example.json` is not (and I allowed `StructureDefinition-example.json` in case the package is malformed)). Let me know what you think.

Reply:
Based on our conversation, I updated how this is implemented in [2ee37fe](https://github.com/FHIR/fhir-package-loader/pull/52/commits/2ee37fe8acd436533459dbcb385b2d8348357d67). This should only check for files in the `package` folder or in the root folder, which aligns with the `cleanCachedPackage` and non-recursive approach in the DiskBasedPackageCache. I tried to explain that in the comment, but let me know if I can make it clearer.

Reply:
It's good! Thank you!